extending/plugins.rst

```diff
@@ -0,0 +1,134 @@
+the ``conanfile`` object) and in that case it checks if the download of the sources are likely coming from immutable places (a compressed
+file or a determined :command:`git checkout`). This is done **before** the **source()** method of the recipe is called.
+
+Any kind of Python scripting can be executed. You can create global functions and called them from different plugin functions, import them
+from a relative module and warn, error or even raise to block the Conan client execution.
+
+As you can see each function receives some parameters but not all of them are available for all functions as this may change depending on
+the context of the commands being executed such as the recipe being in the local cache or not.
+
+.. important::
+
+    A detailed description of the functions allowed and its parameters as well as their execution can be found in it dedicated reference
+    section: :ref:`plugins_reference`.
+
+Other useful task where a plugin may come handy are the upload and download actions. There are **pre** and **post** functions for every
+download/upload as a whole and for fine download task such as recipe and package downloads/uploads.
+
+For example they can be used to sign the packages before being uploaded and checking that signature when they are downloaded.
+
+.. code-block:: python
+   :caption: *signing_plugin.py*
+
+    import os
+    from conans import tools
+
+
+    def pre_upload_package(output, conanfile_path, reference, package_id, remote, **kwargs):
+        package_path = os.path.abspath(os.path.join(os.path.dirname(conanfile_path), "..", "package", package_id, "conan_package.tgz"))
+        my_signing_package_function(package_path)
+
+    def post_download_package(output, conanfile_path, reference, package_id, remote, **kwargs):
+        package_path = os.path.abspath(os.path.join(os.path.dirname(conanfile_path), "..", "package", package_id, "conan_package.tgz"))
+        my_sign_checker_function(package_path)
+
+Official Plugins
+----------------
+
+There are two official plugins ready to be used in Conan. You could take as a starting point to create your own ones.
+
+attribute_checker
++++++++++++++++++
+
+The first one is the *attribute_checker.py* plugin that it is shipped with the Conan client. It has the functionality of warning when
+recipes do not contain some metadata attributes.
+
+.. code-block:: python
+   :caption: *attribute_checker.py*
+
+    def pre_export(output, conanfile, conanfile_path, reference, **kwargs):
+        # Check basic meta-data
+        for field in ["url", "license", "description"]:
+            field_value = getattr(conanfile, field, None)
+            if not field_value:
+                output.warn("Conanfile doesn't have '%s'. It is recommended to add it as attribute"
+                            % field)
+
+This plugin comes activated by default.
+
+Conan Center plugin
```

Comment:
I wouldn't mention the conan center plugin until we really use it and complete it.

Reply:
I'd like to keep it because it could be used as a reference to start creating a plugin. I have added a warning saying this is something under development